### PR TITLE
docs: update options

### PR DESCRIPTION
**Pull request by hacdias**
Related to filebrowser/filebrowser#497 and filebrowser/caddy#4.

### CODE REVIEW

File: configuration.md [hunk 1]
```diff
@@ -18,23 +18,32 @@ This meaning of each flag is transcendant to the correspondant options in the co
...
-- ```--commands``` is a space separated string with the available commands for new users. Defaults to "git svn hg".
-- ```--no-auth``` disables the authentication. Using this option, the default values will be used for the permissions.
-- ```-s, --scope``` is the default scope for new users. Defaults to the working directory.
+- ```--defaults.allowCommands``` is the default value for allow commands option. Defaults to true.
+- ```--defaults.allowEdit``` is the default value for allow edit option. Defaults to true.
+- ```--defaults.allowNew``` is the default value for allow new option. Defaults to true.
+- ```--defaults.commands``` is a space separated string with the available commands for new users. Defaults to "git svn hg".
+- ```-s, --defaults.scope``` is the default scope for new users. Defaults to the working directory.
+
+If `auth.method` is set to "none", then the previous values will be used for the permissions.
```
Comment (1138-4EB) [inline configuration.md]: Maybe reword it to `then the defaults listed above will be applied (see #349)`.

File: configuration.md [hunk 2]
```diff
@@ -18,23 +18,32 @@ This meaning of each flag is transcendant to the correspondant options in the co
...
+ReCAPTCHA specific options:
+
+- ```--recaptcha.host``` is the ReCAPTCHA site host. By default we use Google's, but some users in China might find it more useful to use `https://recaptcha.net`.
+- ```--recaptcha.key``` is the ReCAPTCHA site key. Enables ReCAPTCHA on login.
+- ```--recaptcha.secret``` is the ReCAPTCHA secret key. Enables ReCAPTCHA on login.
+
+Authentication specific options:
+
+- ```--auth.method``` it the method of authentication between 'none', 'default' and 'proxy' authentication. Defaults to "default".
+- ```--auth.header``` is the header name used for proxy authentication. Default to "X-Forwarded-User".
```
Comment (1138-4EB) [inline configuration.md]: Any other reference/explanation is required related to filebrowser/filebrowser#485?

File: configuration.md [hunk 3]
```diff
@@ -18,23 +18,32 @@ This meaning of each flag is transcendant to the correspondant options in the co
...
-- ```-s, --scope``` is the default scope for new users. Defaults to the working directory.
+- ```--defaults.allowCommands``` is the default value for allow commands option. Defaults to true.
+- ```--defaults.allowEdit``` is the default value for allow edit option. Defaults to true.
+- ```--defaults.allowNew``` is the default value for allow new option. Defaults to true.
+- ```--defaults.commands``` is a space separated string with the available commands for new users. Defaults to "git svn hg".
+- ```-s, --defaults.scope``` is the default scope for new users. Defaults to the working directory.
+
+If `auth.method` is set to "none", then the previous values will be used for the permissions.
 
 So, if you wanted to run File Browser on port 80, with the database on `/etc/fm.db` and the default scope to `/data`, you would run:
```
Comment (1138-4EB) [inline configuration.md]: `fm.db` or `fb.db`?

File: configuration.md [hunk 4]
```diff
@@ -63,64 +72,85 @@ Here is a specimen of a JSON configuration file:
...
-  "allowCommands": true,
-  "allowEdit": true,
-  "allowNew": true,
-  "commands": [
-    "git",
-    "svn"
-  ]
+  "auth": {
+    "method": "default",
+    "header": ""
```
Comment (1138-4EB) [inline configuration.md]: Empty fields are required or did you add them to let users know that the option exists?

**Comment by maweck** [comment]
I choose the comment as it is easier for me right now.

I would maybe add a small image like the one below to make the idea more clear.

untitled diagram

If 'auth.method" is set to "proxy" filebrowser will search for the configured 'auth.loginHeader' in the request. In case the header is present, filebrowser will check if it knows a user with the username and will log in this user automatically. If not, the login page is displayed. 

*Warning:*
Filebrowser will fully trust the provided header! In case the reverse-proxy can be bypassed, an attacker could simply attach an "X-Forwarded-Header" to the request to gain admin access.

Is that fine for you?
